vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts

```diff
@@ -21,9 +22,13 @@ export class WebRequestWorker {
         if (isNullOrUndefined(this.cachedData)) {
             // Have to acquire data before continuing
             this.cachedData = await this.makeWebRequest();
-        } else {
+        } else if (isNullOrUndefined(this.currentRequest)) {

```

Comment:
In which case currentRequest is null?

Reply:
Actually what is "currentRequest"? Why `this.makeWebRequest().then((result) => this.cachedData = result);` is blocking? it is using promise, no?

Reply:
It won't be null, this could just as easily be `=== undefined`. Is it generally better style to do that as opposed to `isNullOrUndefined`?

Since we don't await `this.currentRequest.then...`, it's not blocking, `then` just registers the callback and continues execution. This means that `currentRequest` will be defined until that callback is called, eliminating redundant requests.